cloudinit/sources/DataSourceEc2.py

```diff
@@ -317,6 +321,7 @@ def wait_for_metadata_service(self):
                 headers_redact=AWS_TOKEN_REDACT,
                 headers_cb=self._get_headers,
                 request_method=request_method,
+                connect_synchronously=False,
```

Comment:
On EC2 official platform (which is the only Ec2 lookalike that we are currently aware of with IPv6 IMDS support) we should be providing  this connect_synchronously=False only to the call `uhelp.wait_for_url` in `_maybe_fetch_api_token` on line 259.  We shouldn't attempt this here as our first interaction to official EC2 IMDS platform is the latest/api/token which is an "allowed" URL for us to hit/retry multiple times without triggering "non-token IMDS interaction warnings/counts" on EC2 instances.

Reply:
And given that `_maybe_fetch_api_token` is only intended to support EC2 proper platforms, this feels like the right scope for this changeset anyway. If other EC2-lookaline platforms grow IPv6 IMDS support we can adapt in the future.